ARROW-4127: [Documentation][Python] Add instructions to build with Docker

cav71 opened the pull request:
Fix indentation in the dockek instructions (was https://github.com/apache/arrow/pull/3274)

xhochy commented on docs/source/python/development.rst:
```diff
@@ -414,3 +416,21 @@ After these steps are completed, the documentation is rendered in HTML
+  docker-compose run docs 
+
+
+The final output is located under::
```
There is also final output located in `docs/_build/html`. The copy at `site/asf-site/docs/latest` is only relevant for uploading to the website. I would prefer that users look in the `docs/` folder.

cav71 replied:
fixed

xhochy commented on docs/source/python/development.rst:
```diff
@@ -414,3 +416,21 @@ After these steps are completed, the documentation is rendered in HTML
+.. code-block:: shell
+
+  docker-compose build cpp 
+  docker-compose build python 
```
This is missing a `docker-compose build docs`.

cav71 replied:
fixed, thanks. Would possible to speed up the documentation build? Perhaps taking advantage of layering and splitting the cpp/python build?

wesm commented:
@cav71 it is preferable to have a single PR for each change (this is the 3rd PR for this one), and if you need to make changes that you either force-push the branch (which automatically updates the PR) or push new commits. This helps the maintainers stay organized. If you need help with git workflows please don't hesitate to ask

cav71 commented:
Hi @wesm sorry for the confusion, yes I'm new to the git workflows too. If I understand it right I'll have to force push changes into this PR to avoid further confusion: I hope I've done right this time.

cav71 commented:
Hi @wesm,

I've updated building.rst with docker instructions and the development.rst so point to the building.rst so we won't have duplicate documents.

I've rebased to master, but for some reason the PR seems catching also changes from upstream/master. The git diff --name-only upstream/master..origin/documentation shows only building.rst and development.rst as changes (as it should be): does the github computes the difference from master at branch creation? I'm baffled here.

Thanks

wesm commented:
> I've rebased to master, but for some reason the PR seems catching also changes from upstream/master

It looks like you have a mix of rebases and merges in this branch, which is going to cause issues. I'd recommend never using `git merge` for this project unless you are doing `git merge -ff-only`
